pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-common/src/main/java/org/apache/pinot/plugin/ingestion/batch/common/SegmentGenerationTaskRunner.java

```diff
@@ -62,6 +62,10 @@ public class SegmentGenerationTaskRunner implements Serializable {
   public static final String FILE_PATH_PATTERN = "file.path.pattern";
   public static final String SEGMENT_NAME_TEMPLATE = "segment.name.template";
 
+  // For UploadedRealtimeSegmentNameGenerator
+  public static final String SEGMENT_PARTITION_ID = "segment.partitionId";
+  public static final String SEGMENT_UPLOAD_TIME_MS = "segment.uploadTimeMs";
```

Comment:
yeah I see everything else is duplicated here so I also duplicate this one, I can update to use BatchConfigProperties

Reply:
`SegmentGeneratorConfig.segmentCreationTime` can be reused as `SEGMENT_UPLOAD_TIME`. Every run of ingestion job should have a time associated with it and should populate it.

Since Flink can also be run in batch mode, e.g. hourly/daily; start/end time can be used as `creationTime`. This provides deterministic segment names and one run's segments will have same timestamp in segment name.

Flink uses table config but also overrides transient properties like partitionId, seqId, creationTime in [FlinkSegmentWriter](https://github.com/apache/pinot/pull/13837/files#diff-72b89e65ea292a808b5d8a902d8a64adb87834534414d5b9d0e46c4725a0455cL126) through flink job config `SEGMENT_UPLOAD_TIME_MS`. This way fixed properties(like prefix) can be part of table config and transient properties should come from the ingestion job.

Reply:
To use `SegmentGeneratorConfig.segmentCreationTime` you have to call `SegmentGeneratorConfig.setCreationTime()` here first. Are you suggesting calling it here with system time instead of letting user customize it? I don't think we can use `SegmentGeneratorConfig.segmentCreationTime` directly without setting it. 
Spark is quite similar, it can also run in batch mode and one run's segments will have same timestamp in segment name